src/main/c/cext/ruby.c

```diff
@@ -4349,7 +4349,8 @@ VALUE rb_str_cat_cstr(VALUE str, const char *ptr) {
 }
 
 st_index_t rb_hash_start(st_index_t h) {
-  rb_tr_error("rb_hash_start not implemented");
+  st_index_t seed = (st_index_t) polyglot_as_i64(RUBY_CEXT_INVOKE_NO_WRAP("context_hash_seed"));
+  return seed + h;
```

Comment:
Could we rather pass `h` to Ruby and then call `TrufflePrimitive.vm_hash_start(h)`?
(`st_index_t` is `st_data_t` which is `unsigned long`, so `ULONG2NUM(h)` seems a good way).

I think we shouldn't expose a direct way to get the seed (even though with `rb_hash_start` it's easy to get it with the current `vm_hash_start` implementation).
This would then also work if we change the implementation of `vm_hash_start`.